lib/fennec/stun.ex

```diff
@@ -2,16 +2,19 @@ defmodule Fennec.STUN do
   @moduledoc false
   # Processing of STUN messages
 
-  alias Jerboa.Params
+  alias Fennec.TURN
+  alias Fennec.Evaluator
 
-  @spec process_message!(binary, Fennec.ip, Fennec.portn) :: binary | no_return
-  def process_message!(data, ip, port) do
-    {:ok, x = %Params{method: :binding}} = Jerboa.Format.decode(data)
-    case Fennec.Evaluator.service(x, %{address: ip, port: port}) do
+  @spec process_message(binary, Fennec.ip, Fennec.portn, TURN.t) ::
```

Comment:
For me, it's quite simple. It is `STUN` protocol (`TURN` is `STUN` on protocol level). The protocol state is called `turn` to indicate that this state is *only* for `TURN` requests, since base `STUN` is stateless.